packages/components/src/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { ChangeEvent } from "react";
+import classnames from "classnames";
+import styles from "./Checkbox.css";
+import { Icon } from "../Icon";
+import { Text } from "../Text";
+
+interface CheckboxProps {
+  /**
+   * Determines if the checkbox is checked or not.
+   */
+  readonly checked?: boolean;
+
+  /**
+   * Initial checked value of the checkbox. Only use this when you need to
+   * pre-populate the checked attribute that is not controlled by the components
+   * state. If a state is controlling it, use the `checked` prop instead.
+   */
+  readonly defaultChecked?: boolean;
+
+  /**
+   * Disables the checkbox.
+   */
+  readonly disabled?: boolean;
+
+  /**
+   * When `true` the checkbox to appears in indeterminate.
+   *
+   * @default false
+   */
+  readonly indeterminate?: boolean;
+
+  /**
+   * Label that shows up beside the checkbox.
+   */
+  readonly label?: string;
+
+  /**
+   * Checkbox input name
+   */
+  readonly name?: string;
+
+  /**
+   * Value of the checkbox.
+   */
+  readonly value?: string;
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#Value

Reply:
Will this element work in a form as-is or do you have extract the value from the component some other way to submit it?

Reply:
It should work just fine in a form. 😄

Reply:
`<form>` still kinda confuses me. I'm never really sure what behaviour is built in because almost all the places I've worked did something other that just `<form action=...>`